docs/source/etcd-data-model.rst

```diff
@@ -110,6 +118,13 @@ The various properties in this object have the following meanings:
   this endpoint. Each profile is applied to packets in the order that they
   appear in this list.
 
+``ipv4_nat``
+  a list of 1:1 NAT mappings to apply to the endpoint.  Inbound connections to
+  out_ip will be forwarded to in_ip.  Connections initiated from in_ip will
+  not have their source address changed, except when an endpoint attempts to
+  connect one if it's own out_ips.  Each in_ip must be associated with the
```

Comment:
I'm not sure how best to phrase that.  In other SDNs, setting up a floating IP also changes the source IP for connections initiated from the endpoint.  That way, if the floating IP was a public IP with external internet access, and the fixed IP was an internal IP without external internet access, the endpoint would have external internet access.  Generally for internal to internal, the fixied IP is still used.  It is more difficult to decide when to do this in Calico, since in other SDNs there is more of a boarder from internal to external.  Likely if we added support for this, it would be some sort of configuration option to figure out how to decide when to do the SNAT and when not to.  It's complicated enough of a problem that I didn't want to solve it right now.

The exception is I had to change the source address (via a SNAT) when the endpoint tries to connect to it's own floating IP.  If I didn't, the VM would get a packet that said it came from itself, but originated from outside of itself, which would generally break by basic security filtering, and also the connection tracking table would get confused.